pipeline_dp/aggregate_params.py

```diff
@@ -88,33 +92,47 @@ def __post_init__(self):
+                              int) or self.max_partitions_contributed <= 0:
+                raise ValueError(
+                    "params.max_partitions_contributed must be set "
+                    "to a positive integer")
+            if not isinstance(self.max_contributions_per_partition,
+                              int) or self.max_contributions_per_partition <= 0:
+                raise ValueError(
+                    "params.max_contributions_per_partition must be set "
+                    "to a positive integer")
+            if needs_min_max_value and (self.min_value is None or
+                                        self.max_value is None):
+                raise ValueError(
+                    "params.min_value and params.max_value must be set")
+            if needs_min_max_value and (_not_a_proper_number(self.min_value) or
+                                        _not_a_proper_number(self.max_value)):
+                raise ValueError(
+                    "params.min_value and params.max_value must be both finite numbers"
+                )
+            if needs_min_max_value and self.max_value < self.min_value:
+                raise ValueError(
+                    "params.max_value must be equal to or greater than params.min_value"
+                )
+        if self.custom_combiners:
+            logging.warning("Warning: custom combiners are used. This is an "
+                            "experimental feature. It might not work properly "
+                            "and it might be changed orremoved without any "
+                            "notifications.")
+        if self.metrics and self.custom_combiners:
+            # TODO(dvadym): after implementation of custom combiners to verify
+            # whether this check is required?
             raise ValueError(
-                "params.max_contributions_per_partition must be set "
-                "to a positive integer")
-        if needs_min_max_value and (self.min_value is None or
-                                    self.max_value is None):
-            raise ValueError(
-                "params.min_value and params.max_value must be set")
-        if needs_min_max_value and (_not_a_proper_number(self.min_value) or
-                                    _not_a_proper_number(self.max_value)):
-            raise ValueError(
-                "params.min_value and params.max_value must be both finite numbers"
-            )
-        if needs_min_max_value and self.max_value < self.min_value:
-            raise ValueError(
-                "params.max_value must be equal to or greater than params.min_value"
-            )
+                "Custom combiners can not be used with standard metrics")
 
     def __str__(self):
-        return f"Metrics: {[m.value for m in self.metrics]}"
+        if self.metrics:
+            return f"Metrics: {[m.value for m in self.metrics]}"
+        return "Custom metrics."
```

Comment:
if we enable custom combiners to define their human-readable names (see the other comment in the custom combiner definition), we could print that here instead of generic "Custom metric".